### PR TITLE
Zen 438 herkin UI

### PR DESCRIPTION
**Pull request by daniel112**
**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-438)

## Context

* Setting up the initial screens and modal for keg-herkin
* initially Display 3 component as modal ( ‘select test type’ dropdown (gherkin default), `Select test file` dropdown and `Create new file` btn)

## Goal

* Implement the modal with initial selectors to setup which test file user wants to see
* do some cleanup on some function/files

## Updates

* `repos/tap/actions/features/loadFeature.js`
    * will load in the feature file and changes to a specified screen if passed in 
* `repos/tap/actions/features/setFeatureActive.js`
    * cleanup
* `repos/tap/actions/init.js`
   * updated so it displays the init modal if no querystring is passed in
* `repos/tap/actions/modals/*`
   * basic modal actions, similar to the ones on evf
* `repos/tap/components/form/select/select.js`
   * custom select component using keg-components
* `repos/tap/components/modals/testSelectorModal.js`
   * the select modal component
   * user picks which tab, file and test type they want to initialize
* `repos/tap/components/runner/runner.js`
  * removing some unused lines 
* `repos/tap/hooks/useFeature.js`
  * minor adjust to be able to fetch feature based on the feature name
### Setup
* Run `keg herkin pack run package=keg-herkin:zen-438-herkin-ui ports=5005:5005`
* navigate to http://herkin-zen-438-herkin-ui.local.kegdev.xyz/
* Verify that it loads the settings modal initially
  * [image](https://user-images.githubusercontent.com/3317835/107095060-5fe51900-67c5-11eb-9e55-60f09250a8e8.png)
  * 
### Google Search option
* Verify the `Select initial tab:` select contains 3 items: Editor/Builder/Runner. with `Editor` as default
* Verify `Test Type` is only `Gherkin`
* Verify `Select Test File` defaults to `Create new file`
* select `Google Search` as the test file
* click `Start`
* Verify that it navigates to the correct `Tab` and loads the `Google Search` Feature file

### Create new file option
* Refresh the browser to see the Settings modal
* Select `Create New File` 
* Select `Initial Tab`: Runner
* It should console log a placeholder message
* as a placeholder It should drop you in the `Test Runner` tab with the example jest test

### CODE REVIEW

File: repos/tap/components/form/select/select.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { reStyle } from '@keg-hub/re-theme/reStyle'
+
+
+const ReStyleView = reStyle(
+  View,
+  'style'
+)((__, props) => ({
+  ...props?.style,
+  padding: 8,
+}))
```
Comment (mpcarolin) [inline repos/tap/components/form/select/select.js:17]: Sweet. I really like this way of styling components. I like that it separates all the styling logic into a function completely separate from the component, and it still has access to the props for dynamic stuff.

File: repos/tap/components/form/select/select.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { 
+  Select as KegSelect, 
+  Option, 
+  Label,
+  View
+} from '@keg-hub/keg-components'
+import { reStyle } from '@keg-hub/re-theme/reStyle'
+
+
+const ReStyleView = reStyle(
```
Comment (lancetipton) [inline repos/tap/components/form/select/select.js:11]: Ahh snap... breaking out the reStyle. I love it

File: repos/tap/components/modals/testSelectorModal.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  const loadTests = useCallback(() => {
+
+    testName === Values.CREATE_NEW_FILE
+      ? createFeatureFile(selectedTab)
+      : loadFeature(feature, selectedTab)
+
+      setModalVisibility(false)
+  }, 
+  [
+    Values.CREATE_NEW_FILE,
```
Comment (mpcarolin) [inline repos/tap/components/modals/testSelectorModal.js:91]: this is a constant, right? So I think we can safely omit it from the dependencies.

File: repos/tap/containers/screens/screen.js [hunk 4]
```diff
@@ -34,24 +34,23 @@ const tabs = [
...
-}, [screen])
+const useScreen = id => useMemo(() => {
+  return tabs.find(item => item.id === id) || tabs[2]
+}, [id])
 
 export const Screen = props => {
 
   const theme = useTheme()
-  const { activeData } = useSelector(({ items }) => pickKeys(
+  const { activeTab } = useSelector(({ items }) => pickKeys(
```
Comment (mpcarolin) [inline repos/tap/containers/screens/screen.js:44]: seems like this could be `useStoreItems` like you used before

**Review by lancetipton — APPROVED** [review]
Looks great man. Awesome work. Thanks

**Review by mpcarolin — APPROVED** [review]
Looks good. Left a few recommendations.

File: repos/tap/components/form/select/select.js (hunk 1 above)
Comment (lancetipton) [reply repos/tap/components/form/select/select.js:17]: Yah, ReStyle is the future

File: repos/tap/components/modals/testSelectorModal.js (hunk 3 above)
Comment (daniel112) [reply repos/tap/components/modals/testSelectorModal.js:91]: true

**Pull request merged**